src/dbus_fast/message_bus.py

```diff
@@ -722,6 +722,13 @@ def _setup_socket(self) -> None:
                 raise InvalidAddressError(f"got unknown address transport: {transport}")
 
         if err:
+            self._fd = None
+            if self._stream:
+                self._stream.close()
+            self._stream = None
+            if self._sock:
+                self._sock.close()
+            self._sock = None
```

Comment:
I think a `contextlib.ExitStack` would be a cleaner way to handle the cleaunup here. Just call `stack.pop_all()` to avoid the cleanup callbacks on a successful return.

Plus it would also handle other error cases where an exception is raised earlier.